ERCS/erc-7715.md

```diff
@@ -0,0 +1,621 @@
+
+### `wallet_grantPermissions`
+
+We introduce a `wallet_grantPermissions` method for the DApp to request the Wallet to grant permissions.
+
+#### Permissions & Policies
+
+Before we specify the JSON-RPC request and response it's important to clarify the relationship between permissions and policies.
+
+Permissions represent a singular piece of authority that can be used by the application. The data restricts the scope of the ability authorized.
+eg. ability to transfer a token, update an ENS domain, vote on a DAO, etc
+
+Policies represent a generalized scope which can be attributed to any given permission. A policy without a permission could be excessively permissive.
+eg. a token allowance, a gas limit, an access list, an ENS record, etc.
+
+For each permission there will be a set of policies and for each policy there will be a set of properties which are specified as follows:
+
+```typescript
+interface Permission {
+  type: name;
+  policies: Policy[];
+  required: boolean;
+  data: any;
+}
+
+interface Policy {
+  type: name;
+  data: any;
+}
+```
+
+#### Request Specification
+
+```tsx
+type GrantPermissionsRequestParams = {
+  account?: `0x${string}`;
+
+  chainId: number;
+
+  signer: {
+    type: string;
+    data: any;
+  };
+
+  permissions: Permission[];
+
+  expiry: number;
+};
+```
+
+`account` is an optional field that identifies the account being targetted for this permissions request which is useful when a connection has been established and multiple accounts have been exposed.
+
+`chainId` is field that identifies that chain with [EIP-155](./eip-155.md) which these permissions are contextual and all addressed can be found for the accounts and permissions.
+
+`signer` is a field that identifies the key or account associated with the session or alternatively the wallet will manage the session. See the “Wallet-managed Sessions” section for details.
+
+`permissions` is a list of permissions with determined policies. See “Policy Types” section for details.
+
+Each permission can be either `required` or not. If a permission is `required`, the wallet MUST grant the permission if it approves the request. Otherwise, the wallet may reject the permission even if it approves the request (presumably because it grants other permissions).
+
+`expiry` is a UNIX timestamp (in seconds) that specifies the time by which this session MUST expire.
+
+Request example:
+
+```tsx
+{
+  account: '0x...'
+
+  chainId: 123,
+
+  signer: {
+    type: 'key',
+    data: {
+      id:'did:key:zQ3shokFTS3brHcDQrn82RUDfCZESWL1ZdCEJwekUDPQiYBme',
+    }
+  },
+
+  permissions: [
+    {
+      type: 'native-token-transfer',
+      policies: [
+        {
+          type: 'gas-limit',
+          data: {
+            limit: '0x...',
+          },
+        },
+        {
+          type: 'call-limit',
+          data: {
+            count: 1,
+          },
+        },
+      ],
+      required: true,
+      data: {
+        address: '0x...',
```

Comment:
I think there shouldn't be `address` field in `native-token-transfer` permission.